docs/sphinx/examples/python/dynamics/ion_trap.py

```diff
@@ -0,0 +1,129 @@
+"""
+Tutorial: Creating a GHZ State with Trapped Ions
+
+This example shows how to prepare a GHZ (Greenberger-Horne-Zeilinger) state 
+using trapped ions and CUDA-Q. We'll use the effective spin model from the 
+famous Sørensen-Mølmer paper.
+
+What we're doing:
+- Start with N ions all in ground state |gg...g⟩
+- Apply an effective Hamiltonian H = 4χJ_x²  
+- Watch the system evolve into a GHZ superposition: (|gg...g⟩ + |ee...e⟩)/√2
+
+This tutorial uses the simplified effective model.
+"""
+
+import cudaq
+from cudaq import spin, Schedule, RungeKuttaIntegrator
+import numpy as np
+import cupy as cp
+import matplotlib.pyplot as plt
+
+cudaq.set_target("dynamics")
+
+# Physical parameters, these come from the Sørensen-Mølmer paper
+nu = 1.0        # Trap frequency (our reference)
+delta = 0.9 * nu    # Laser detuning  
+Omega = 0.1 * nu    # Laser strength
+eta = 0.1           # How strongly lasers couple to motion
+
+# The effective coupling strength (this is the key parameter!)
+chi = (eta**2 * Omega**2 * nu) / (2 * (nu**2 - delta**2))
+
+N = 8                           # Number of ions (start small!)
+evolution_time = 1.0 / chi      # How long to evolve
+num_steps = 100                 # Time resolution
+
+dimensions = {i: 2 for i in range(N)}  # Each ion is a 2-level system
+
+# collective spin operator J_x
+J_x = spin.empty()
+for i in range(N):
+    J_x += spin.x(i)
+J_x /= 2  # Normalize
+
+hamiltonian = 4 * chi * J_x * J_x
+
+# Set up initial state and time evolution  
+# Start with all ions in ground state |gg...g⟩
+initial_state_vector = np.zeros(2**N, dtype=np.complex128)
+initial_state_vector[0] = 1.0  # |00...0⟩ = |gg...g⟩
+initial_state = cudaq.State.from_data(cp.array(initial_state_vector))
```

Comment:
nit: we can construct the cupy array directly to prevent host allocation:

```suggestion
initial_state_vector = cp.zeros(2**N, dtype=cp.complex128)
initial_state_vector[0] = 1.0  # |00...0⟩ = |gg...g⟩
initial_state = cudaq.State.from_data(initial_state_vector)
```